enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,893 @@
+--dst-cert-dir=/my/certs \
+docker://quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:... \
+docker://localhost:12345/openshift-release-dev/ocp-v4.0-art-dev@...
+```
+
+When all the images have been copied to the temporary image registry server it
+will be shutdown.
+
+The result will be a directory containing approximately 180 images and requires
+16 GiB of space. The command will then create a `upgrade-4.13.7-x86_64.tar` tar
+file containing that directory and a `metadata.json` file.
+
+The `metadata.json` file will contain additional information, in particular the
+architecture, the version, the size and the list of images:
+
+```json
+{
+  "version": "4.13.7",
+  "arch": "x86_64",
+  "size": "16 GiB",
+  "release": "quay.io/openshift-release-dev/ocp-release@sha256:...",
+  "images": [
+    "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+    "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+    ...
+  ]
+}
+```
+
+The `metadata.json` file will always be the first entry of the tar file, to
+simplify operations that need the metadata but not the rest of the contents of
+the tar file.
+
+The command will also write a `upgrade-4.13.7-x86_64.sha256` file containing a
+digest of the complete tar file.
+
+The engineer will write the tar file to some kind of media and hand it over to
+the technicians, together with the documentation explaining how to use it.
+
+#### Documentation to use the bundle
+
+This documentation shouldn't assume previous OpenShift knowledge, and should be
+basic instructions to plug the USB stick containing the bundle and then check if
+upgrade the upgrade succeeded or failed.
+
+#### CVO will need to orchestrate the upgrade
+
+The cluster version operator will be responsible for orchestrating the upgrade.
+
+The technician will receive the USB stick containing the
+`upgrade-4.13.7-x86_64.tar` file and will plug it in one of the nodes of the
+cluster.
+
+The cluster version operator will be monitoring device events in all the nodes
+of the cluster in order to detect when such an USB stick (or any other kind of
+media containing an upgrade bundle) has been plugged. To control that a new
+`spec.desiredUpdate.bundle.monitor` field will be added to the `ClusterVersion`
+object. The default value will be `false` so that this monitoring will be
+disabled by default. When set to `true` the cluster version operator will create
+a new `bundle-monitor` daemon set that will perform the actual monitoring. When
+any of the pods in this daemon set detects a device containing a valid upgrade
+bundle it will update the status of the `ClusterVersion` object indicating that
+the bundle is available, via a new `status.desired.bundle.file` field.
+
+For situations where it isn't desirable to monitor device events, or it isn't
+possible to plug the USB stick or any other kind of media, we will also add a
+new `spec.desiredUpdate.bundle.file` field to explicitly indicate the location
```

Comment:
If the technician doing the upgrade is not familiar with OpenShift, who sets this field?

Reply:
In this particular case, when it isn't possible to just plug the USB stick, the technician will need to know a bit more about OpenShift or have detailed instructions. Or else the administrator of the cluster will set this in advance.